casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -366,39 +371,35 @@ object BlockAPI {
       timestamp: Long
   ): F[Option[BlockMessage]] =
     blockHashes.toStream
-      .traverse(ProtoUtil.unsafeGetBlock[F](_))
+      .traverse(ProtoUtil.unsafeGetBlock[F])
       .map(blocks => blocks.find(ProtoUtil.containsDeploy(_, user, timestamp)))
 
   def showBlock[F[_]: Monad: MultiParentCasperRef: Log: SafetyOracle: BlockStore](
       q: BlockQuery
-  ): F[BlockQueryResponse] = {
+  ): Effect[F, BlockQueryResponse] = {
 
     val errorMessage =
       "Could not show block, casper instance was not available yet."
 
-    def casperResponse(implicit casper: MultiParentCasper[F]) =
+    def casperResponse(
+        implicit casper: MultiParentCasper[F]
+    ): Effect[F, BlockQueryResponse] =
       for {
         dag        <- MultiParentCasper[F].blockDag
         maybeBlock <- getBlock[F](q, dag)
         blockQueryResponse <- maybeBlock match {
                                case Some(block) =>
                                  for {
                                    blockInfo <- getFullBlockInfo[F](block)
-                                 } yield
-                                   BlockQueryResponse(
-                                     status = "Success",
-                                     blockInfo = Some(blockInfo)
-                                   )
+                                 } yield BlockQueryResponse(blockInfo = Some(blockInfo)).asRight
```

Comment:
Is there a reason we are getting rid of status? I guess it isn't that useful in the first place.

Reply:
@KentShikama That is the whole point of this PR. We want to use `Either` message to signal errors instead of a `status` field.